### PR TITLE
Set critical logs to "CRITICAL"

### PR DESCRIPTION
**Pull request by Jongy**
1. CRITICAL logs are bad errors that prevent the profiler from operating correctly
2. ERROR logs are stuff we don't expect to happen and might indicate a bug on our code
3. WARN logs are errors we are aware of and can happen

Currently marked 2 bad errors as CRITICAL, more will be added

### CODE REVIEW

File: gprofiler/profilers/factory.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def get_profilers(
         except Exception:
             if profiler_config.profiler_class is SystemProfiler:
                 raise SystemProfilerInitFailure("Could not create the system profiler")
-            logger.exception(f"Couldn't create the {profiler_name} profiler, continuing without it")
+            logger.critical(f"Couldn't create the {profiler_name} profiler, continuing without it", exc_info=True)
```
Comment (d3dave) [inline gprofiler/profilers/factory.py:45]: This is the important one, why don't we exit with error on this, suggesting to run with `--no-{profiler_name}` if user wants to ignore?

File: gprofiler/profilers/perf.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def start(self) -> None:
         except TimeoutError:
             process.kill()
             assert process.stdout is not None and process.stderr is not None
-            logger.error(f"perf failed to start. stdout {process.stdout.read()!r} stderr {process.stderr.read()!r}")
+            logger.critical(f"perf failed to start. stdout {process.stdout.read()!r} stderr {process.stderr.read()!r}")
```
Comment (Jongy) [inline gprofiler/profilers/perf.py:83]: Same for this one, to be honest: https://github.com/Granulate/gprofiler/pull/388/files#r906863849

File: gprofiler/profilers/factory.py (hunk 1 above)
Comment (Jongy) [reply gprofiler/profilers/factory.py:45]: Agree, it should exit here probably. Will change

Comment (Jongy) [reply gprofiler/profilers/factory.py]: Done

File: gprofiler/profilers/perf.py (hunk 2 above)
Comment (Jongy) [reply gprofiler/profilers/perf.py:83]: Done

Comment (Jongy) [reply gprofiler/profilers/perf.py:83]: Although I didn't exit brutally, that's a decision for another PR

**Review by d3dave — APPROVED** [review]

**Pull request merged**